test/promql/tests/resource-min-5m.yaml

```diff
@@ -0,0 +1,134 @@
+      - series: 'kube_pod_container_resource_limits{resource="r1", namespace="ns1", pod="ns1p1", container="ns1p1c1"}'
+        values: '1x10'
+      - series: 'container_last_seen{namespace="ns1", pod="ns1p1", container="ns1p1c1"}'
+        values: '_ _ _ 200+60x7'
+      - series: 'kube_pod_labels{namespace="ns1", pod="ns1p1"}'
+        values: '1x10'
+
+    promql_expr_test:
+      - expr: |
+          sum_over_time(appstudio_container_resource_minutes_gauge{resource="r1", namespace="ns1"}[10m])
+        eval_time: 10m
+        exp_samples:
+          - labels: '{resource="r1", namespace="ns1", pod="ns1p1", container="ns1p1c1",}'
+            value: 7.5
+
+  # container created before the time window and dies before it
+  - interval: 1m
+    input_series:
+      - series: 'kube_pod_container_resource_limits{resource="r1", namespace="ns1", pod="ns1p1", container="ns1p1c1"}'
+        values: '1x10'
+      - series: 'container_last_seen{namespace="ns1", pod="ns1p1", container="ns1p1c1"}'
+        values: '200+60x7 _ _ _'
+      - series: 'kube_pod_labels{namespace="ns1", pod="ns1p1"}'
+        values: '1x10'
+
+    promql_expr_test:
+      - expr: |
+          sum_over_time(appstudio_container_resource_minutes_gauge{resource="r1", namespace="ns1"}[10m])
+        eval_time: 10m
+        exp_samples:
+          - labels: '{resource="r1", namespace="ns1", pod="ns1p1", container="ns1p1c1",}'
+            value: 7.5
+
+  # container created within the time window and dies before it
+  - interval: 1m
+    input_series:
+      - series: 'kube_pod_container_resource_limits{resource="r1", namespace="ns1", pod="ns1p1", container="ns1p1c1"}'
+        values: '1x10'
+      - series: 'container_last_seen{namespace="ns1", pod="ns1p1", container="ns1p1c1"}'
+        values: '_ 200+60x6 _ _ _'
+      - series: 'kube_pod_labels{namespace="ns1", pod="ns1p1"}'
+        values: '1x10'
+
+    promql_expr_test:
+      - expr: |
+          sum_over_time(appstudio_container_resource_minutes_gauge{resource="r1", namespace="ns1"}[10m])
+        eval_time: 10m
+        exp_samples:
+          - labels: '{resource="r1", namespace="ns1", pod="ns1p1", container="ns1p1c1",}'
+            value: 7.5
+
+  # measurement not on the 5m mark - in this case we may lose part of the window
+  - interval: 1m
+    input_series:
+      - series: 'kube_pod_container_resource_limits{resource="r1", namespace="ns1", pod="ns1p1", container="ns1p1c1"}'
+        values: '1x12'
+      - series: 'container_last_seen{namespace="ns1", pod="ns1p1", container="ns1p1c1"}'
+        values: '200+60x12'
+      - series: 'kube_pod_labels{namespace="ns1", pod="ns1p1"}'
+        values: '1x12'
+
+    promql_expr_test:
+      - expr: |
+          sum_over_time(appstudio_container_resource_minutes_gauge{resource="r1", namespace="ns1"}[12m])
+        eval_time: 12m
+        exp_samples:
+          - labels: '{resource="r1", namespace="ns1", pod="ns1p1", container="ns1p1c1",}'
+            value: 10
```

Comment:
@amisstea having the rule summing every 5m, we lose some granularity. Namely, we're going to have the same results for every time window that contains the same set of measurements.
```
_ _ _ _ | _ _ _ _ | _ _ _ _ <-- recording intervals
[ _ _ _ | _ _ _ _ | _ _ _ ]  <-- consumer sums over wider window
_ _ _ [ | _ _ _ _ | ] _ _ _ <-- consumer sums over smaller window provides same results
```

Reply:
Should we make the rule record every 1m?

Reply:
I opened a ticket for this one.